### PR TITLE
Support OpenShift monitoring deployment/removal with scripts

### PR DESCRIPTION
**Pull request by BryanEllington**

### CODE REVIEW

File: monitoring/bin/remove_monitoring_openshift.sh [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+log_info "Removing CA bundle..."
+kubectl delete --ignore-not-found cm grafana-trusted-ca-bundle
+
+log_info "Removing dashboards..."
+monitoring/bin/remove_dashboards.sh
+
+log_info "Removing service monitors..."
+kubectl delete --ignore-not-found podmonitor -n $MON_NS eventrouter
+kubectl delete --ignore-not-found servicemonitor -n $MON_NS elasticsearch
```
Comment (gsmith-sas) [inline monitoring/bin/remove_monitoring_openshift.sh:58]: Service Monitors are now deployed (and removed) from the logging deployment/removal scripts.  Do we need these here too?

File: monitoring/openshift/grafana-proxy-secret.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  labels:
+    k8s-app: grafana
+  name: grafana-proxy
+type: Opaque
+data:
+  session_secret: djRtLW9wZW5zaGlmdC1tb25pdG9yaW5nLXNlY3JldAo=
```
Comment (gsmith-sas) [inline monitoring/openshift/grafana-proxy-secret.yaml:9]: Is this the "real" session secret?  Just a place-holder?

File: monitoring/openshift/grafana-trusted-ca-bundle.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  creationTimestamp: "2021-05-08T01:44:45Z"
+  labels:
```
Comment (gsmith-sas) [inline monitoring/openshift/grafana-trusted-ca-bundle.yaml:5]: Do we need this particular creationTimestamp value (or some hard-coded value)?

**Review by gsmith-sas — COMMENTED** [review]
A couple of minor comments/questions.

File: bin/openshift-include.sh [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+# Assumes bin/common.sh has been sourced
+
+# Detect if it is an OpenShift cluster
+OPENSHIFT_CLUSTER=${OPENSHIFT_CLUSTER:-false}
+if kubectl get ns openshift 2>/dev/null 1>&2; then
+  log_debug "OpenShift detected"
+  OPENSHIFT_CLUSTER=true 
+else
+  log_debug "OpenShift not detected. Skipping 'oc' checks."
+fi
```
[the remark this answers is not in the document]
Comment (BryanEllington) [reply bin/openshift-include.sh:14]: Not sure what the problem is. The oc check only occurs if OPENSHIFT_CLUSTER==true and OPENSHIFT_OC_CHECK==true. Unset or 'false' result in the same behavior.

[the remark this answers is not in the document]
Comment (gsmith-sas) [reply bin/openshift-include.sh:14]: It's a nit but: if someone explicitly set OPENSHIFT_CLUSTER to true but was NOT running on OpenShift, the debug message would indicate that the oc check would be skipped...except that wouldn't happen.  You aren't setting OPENSHIFT_CLUSTER to "false" if you determine you are not running on OpenShift, so nothing is skipped.  The message indicates some change of execution logic will occur due to the failed kubectl call but nothing in the code changes the execution logic.  If you set OPENSHIFT_OC_CHECK to false in the else block, you would be explicitly skipping the oc check.  So, as written, the first part of the debug message is correct, the 2nd is misleading or inaccurate.  But, as I said, this is all a bit of nit.

**Review by gsmith-sas — APPROVED** [review]
LGTM

**Review by davidbass13 — APPROVED** [review]

**Pull request merged**